lib/tapioca/generators/base.rb

```diff
@@ -10,13 +10,18 @@ class Base
       extend T::Sig
       extend T::Helpers
 
+      class FileWriter < Thor
+        include Thor::Actions
+      end
+
       # TODO: Remove me when logging logic has been abstracted
       include Thor::Base
 
       abstract!
 
       sig { params(default_command: String).void }
       def initialize(default_command:)
```

Comment:
I'm not sure that would serve any significant use case for now. That basically takes us back to the previous implementation, just moving the writer to a nested class.

Reply:
It allows you to pass a custom `FileWriter`, for example for unit testing.

Reply:
It will make it easier to test these classes in isolation, though. In tests, we can pass in a fake `FileWriter` and assert operations performed after the call, without even touching the file system.